rholang/src/test/scala/coop/rchain/rholang/interpreter/RegistrySpec.scala

```diff
@@ -0,0 +1,641 @@
+                       ListChannelWithRandom(Seq(Quote(GInt(8))), result0Rand),
+                       false)),
+        List()
+      )))
+    result.get(resultChanList("result1")) should be(
+      Some(Row(
+        List(
+          Datum.create(Channel(Quote(GString("result1"))),
+                       ListChannelWithRandom(Seq(Quote(GInt(9))), result1Rand),
+                       false)),
+        List()
+      )))
+    result.get(resultChanList("result2")) should be(
+      Some(Row(
+        List(
+          Datum.create(Channel(Quote(GString("result2"))),
+                       ListChannelWithRandom(Seq(Quote(GInt(10))), result2Rand),
+                       false)),
+        List()
+      )))
+    result.get(resultChanList("result3")) should be(
+      Some(Row(
+        List(
+          Datum.create(Channel(Quote(GString("result3"))),
+                       ListChannelWithRandom(Seq(Quote(GInt(8))), result3Rand),
+                       false)),
+        List()
+      )))
+    result.get(resultChanList("result4")) should be(
+      Some(Row(
+        List(
+          Datum.create(Channel(Quote(GString("result4"))),
+                       ListChannelWithRandom(Seq(Quote(GInt(9))), result4Rand),
+                       false)),
+        List()
+      )))
+    result.get(resultChanList("result5")) should be(
+      Some(Row(
+        List(
+          Datum.create(Channel(Quote(GString("result5"))),
+                       ListChannelWithRandom(Seq(Quote(GInt(8))), result5Rand),
+                       false)),
+        List()
+      )))
+    result.get(resultChanList("result6")) should be(
+      Some(Row(
+        List(
+          Datum.create(Channel(Quote(GString("result6"))),
+                       ListChannelWithRandom(Seq(Quote(GInt(8))), result6Rand),
+                       false)),
+        List()
+      )))
+    result.get(List[Channel](Quote(Registry.registryRoot))) should be(
+      Some(Row(
+        List(
+          Datum.create(Channel(Quote(Registry.registryRoot)),
+                       ListChannelWithRandom(Seq(Quote(EMapBody(ParMap(SortedParMap.empty)))), rootRand),
+                       false)),
+        List()
+      )))
+  }
+
+  "Public lookup" should "decode and then call lookup" in {
+    val lookupString = """
+      new r(`rho:registry:lookup`) in {
+        r!(`rho:id:bnm61w3958nhr5u6wx9yx6c4js77hcxmftc9o1yo4y9yxdu7g8bnq3`, "result0") |
+        r!(`rho:id:bnmzm3i5h5hj8qyoh3ubmbu57zuqn56xrk175bw5sf6kook9bq8ny3`, "result1")
+      }"""
+    val lookupPar: Par = Interpreter.buildNormalizedTerm(new StringReader(lookupString)).value
+    val completePar                     = lookupPar.addSends(rootSend, branchSend)
+    implicit val rand: Blake2b512Random = baseRand.splitByte(4)
+    val newRand = rand.splitByte(2)
+    val randResult0                     = newRand.splitByte(0)
+    randResult0.next; randResult0.next
+    val randResult1 = newRand.splitByte(1)
+    randResult1.next; randResult1.next
```

Comment:
This is sad that you need to do that dance with the PRNG in this test. `rho:registry:lookup` is high level interface.